### DISCUSSION TITLE
shout  output streaming fails after hours without notice.

### DISCUSSION THREAD

**Discussion by SebaLedesma** [MusicPlayerDaemon/MPD#1446] [Q&A]
I'm tracking a problem with shout output plugin.
I was using mpd 0.20.18-1build1 on Ubuntu 18:04, uname -a reports:
` Linux myOdroid-c2 3.16.81-49 #1 SMP PREEMPT Wed Jan 15 21:38:53 -02 2020 aarch64 aarch64 aarch64 GNU/Linux
` The hardware it's an Odroid C2.
 
 Now we are switching to Odroid C4 wich uses Ubuntu 20.04:
` Linux myOdroid-c4 4.9.277-78 #1 SMP PREEMPT Tue Dec 28 22:30:16 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux`
 mpd it's 0.21.20-1build1
 
My mpd.conf it's;
```
...
audio_output {
  type            "shout"
  enabled         "yes"
  protocol        "icecast2"
  host            "radios.streamingdha.com.ar"
  port            "8040"
  mount           "/stream"
#  user            "sebas"   # optional
  password      "mypass"
  encoding      "mp3"
#  encoder      ""
  name          "myProgram"
  description   "my streaming"
#  quality      "5.0"
  bitrate       "320"
  format        "44100:16:2"
  url           "http://www.sebas.com"
#  genre         "jazz"                 # optional
  public        "yes"                   # optional
#  timeout      "2"                     # optional
  mixer_type    "software"              # optional
}
```

I've tested with Icecast-2 and Shoutcast-2 servers. Initially it works fine, but after a day the server loses the stream.
- mpd it's running.
- mpc reports:
```
alsa://pulse
[playing] #1/1 2180:27/0:00 (0%)
volume:100%   repeat: on    random: off   single: off   consume: off
```
- mpc outputs
`Output 1 (myProgram) is enabled`

- The ~/.mpd/state file shows:
```
sw_volume: 100
audio_device_state:1:myProgram
state: play
current: 0
time: 0.978000
random: 0
repeat: 1
single: 0
consume: 0
crossfade: 0
mixrampdb: 0.000000
mixrampdelay: -1.000000
playlist_begin
song_begin: alsa://pulse
song_end
playlist_end

```
- mpd.log dont report nothing, even with log_level "verbose".

So, everything seems fine but there is no stream at the server.
mpc disable 1
mpc enable 1
solves the problem, but I cannot doing this blindly.

Any ideas?
Thanks in advance.

**Comment by MaxKellermann** [comment]
Your MPD version is very old and unsupported.

**Comment by SebaLedesma** [comment]
Thanks, as it's the most updated version that my distro provides I will try to build myself from git repository.

**Reply by SebaLedesma** [reply]
I was able to build mpd 0.22.11 and now I'm testing it.
I can't built 0.23.x as it requires meson 0.56 and my distro provides 0.53.2

**Reply by SebaLedesma** [reply]
mpd 0.22.11 also silently fails on 
Linux mySystem 4.9.277-78 #1 SMP PREEMPT Tue Dec 28 22:30:16 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux
after a couple of hours of running.
mpd reports:
```
alsa://pulse
[playing] #1/1 1622:41/0:00 (0%)
volume:100%   repeat: on    random: off   single: off   consume: off
```
mpc outputs:
`Output 1 (myOutput) is enabled`
and the "state" file also reports the output in play.

In the other hand, 
`netstat -ano | grep 8040`
dont report anything.

Any ideas?
Thanks in advance
